From 3da2447ff92e53d1bf82147a6e3b0af5b20b2bb4 Mon Sep 17 00:00:00 2001
From: Vivek Goyal <vgoyal@redhat.com>
Date: Tue, 19 Aug 2014 22:02:51 -0400
Subject: [crypto] pefile: Validate PKCS#7 trust chain

Message-id: <1408485785-777-20-git-send-email-vgoyal@redhat.com>
Patchwork-id: 88372
O-Subject: [RHEL7.1 PATCH 19/33] pefile: Validate PKCS#7 trust chain
Bugzilla: 1129779
RH-Acked-by: David Howells <dhowells@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

BZ:1129779

commit 98801c002f7e573b4a86bcd5b234864d375e98a0
Author: David Howells <dhowells@redhat.com>
Date: Tue, 1 Jul 2014 16:02:52 +0100

Validate the PKCS#7 trust chain against the contents of the system keyring.

Signed-off-by: David Howells <dhowells@redhat.com>
Acked-by: Vivek Goyal <vgoyal@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/crypto/asymmetric_keys/verify_pefile.c b/crypto/asymmetric_keys/verify_pefile.c
index 029a365..79175e6 100644
--- a/crypto/asymmetric_keys/verify_pefile.c
+++ b/crypto/asymmetric_keys/verify_pefile.c
@@ -449,7 +449,7 @@ int verify_pefile_signature(const void *pebuf, unsigned pelen,
  if (ret < 0)
   goto error;
 
- ret = -ENOANO; // Not yet complete
+ ret = pkcs7_validate_trust(pkcs7, trusted_keyring, _trusted);
 
 error:
  pkcs7_free_message(ctx.pkcs7);
-- 
1.7.1